staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/envelope.go

```diff
@@ -167,6 +179,9 @@ func (t *envelopeTransformer) TransformToStorage(ctx context.Context, data []byt
 		klog.V(2).InfoS("observed different key IDs when encrypting content using kms v2 envelope service", "uid", uid, "objectKeyID", encObject.KeyID, "statusKeyID", statusKeyID)
 	}
 
+	// It's possible to record empty keyID
+	metrics.RecordKeyID(metrics.ToStorageLabel, metrics.LastStatusKeyIDLabel, t.providerName, statusKeyID)
```

Comment:
Discussing this with @enj, we probably want to move this metric to the `probe.check()` call so it's done only once when the `Status` RPC call is performed.

Reply:
To record keyID returned by the probe, I think it makes more sense to track the last timestamp than counter since it's going to be called every minute. WDYT? If so, then I can create a separate metric for that. e.g.

> apiserver_envelope_encryption_keyidhash_status_last_timestamp_seconds{key_id_hash="sha256", provider_name="providerName"} 1.674865558833728e+09

Reply:
Done. Added `apiserver_envelope_encryption_key_id_hash_status_last_timestamp_seconds{key_id_hash="sha256", provider_name="providerName"} 1.674865558833728e+09` and `type ProbeHealthzCheckFunc func(context.Context) (err error)` PTAL